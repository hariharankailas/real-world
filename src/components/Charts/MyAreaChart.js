/***********************************************************************************************
 * File : MyAreaChart
 * Desc : Out of the box Area chart from recharts js
 * Author: Hariharan Kailas
 * Date: 10-Jul-2021
 ***********************************************************************************************/
import React from "react";
import {
  AreaChart,
  ResponsiveContainer,
  Area,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip
} from "recharts";

const MyAreaChart = ({ performanceData }) => {
  return (
          <ResponsiveContainer width="100%" height={400}>
    <AreaChart
      width={500}
      height={400}
      data={performanceData}
      margin={{
        top: 10,
        right: 30,
        left: 0,
        bottom: 0
      }}
    >
      <CartesianGrid strokeDasharray="3 3" />
      <XAxis dataKey="name" />
      <YAxis />
      <Tooltip />
      <Area type="monotone" dataKey="duration" stroke="#8884d8" fill="#8884d8" />
      </AreaChart>
      </ResponsiveContainer>
  );
 }

export default MyAreaChart;
