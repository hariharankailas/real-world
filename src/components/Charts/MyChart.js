/***********************************************************************************************
 * File : MyChart
 * Desc : Out of the box Line chart from recharts js
 * Author: Hariharan Kailas
 * Date: 10-Jul-2021
 ***********************************************************************************************/
import React from 'react';
import {
  ResponsiveContainer,
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend
} from "recharts";


const MyChart = ({ performanceData }) => {
  return (
  <ResponsiveContainer width="100%" height={400}>
    <LineChart
      width={500}
      height={300}
      data={performanceData}
      margin={{
        top: 5,
        right: 30,
        left: 20,
        bottom: 5
      }}
    >
      <CartesianGrid strokeDasharray="3 3" />
      <XAxis dataKey="name" />
      <YAxis />
      <Tooltip />
      <Legend />
      {/* <Line
        type="monotone"
        dataKey="pv"
        stroke="#8884d8"
        activeDot={{ r: 8 }}
      /> */}
      <Line type="monotone" dataKey="duration" stroke="#82ca9d" />
    </LineChart>
</ResponsiveContainer>
  );
 }

export default MyChart;

