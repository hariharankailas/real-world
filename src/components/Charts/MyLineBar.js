/***********************************************************************************************
 * File : MyLineBar
 * Desc : Out of the box Line Bar chart from recharts js
 * Author: Hariharan Kailas
 * Date: 10-Jul-2021
 ***********************************************************************************************/
import React from "react";
import {
  ResponsiveContainer,
  ComposedChart,
  Line,
  Bar,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend
} from "recharts";


const MyLineBar = ({ performanceData }) => {
  return (
      <ResponsiveContainer width="100%" height={400}>
    <ComposedChart
      width={500}
      height={400}
      data={performanceData}
      margin={{
        top: 20,
        right: 20,
        bottom: 20,
        left: 20
      }}
    >
      <CartesianGrid stroke="#f5f5f5" />
      <XAxis dataKey="name" scale="band" />
      <YAxis />
      <Tooltip />
      <Legend />
      <Bar dataKey="duration" barSize={20} fill="#413ea0" />
      <Line type="monotone" dataKey="duration" stroke="#ff7300" />
      </ComposedChart>
      </ResponsiveContainer>
  );
}

export default MyLineBar;