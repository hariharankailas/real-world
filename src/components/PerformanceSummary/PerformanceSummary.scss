.performance-summary-container {
  margin: 2rem auto;
  text-align: center;
}

.performance-summary-chart {
  display: flex;
  justify-content: center;
  margin: 3rem 0;
  min-height: 500px;
  font-size: 1.25rem;
}

.performance-summary-toggle-chart {
  display: flex;
  justify-content: center;
  .btn-group {
    width: auto;
  }
}

@media (max-width: 767.98px) {
  .performance-summary-container {
    margin: 2rem auto;
    text-align: center;
  }

  .performance-summary-chart {
    display: flex;
    justify-content: center;
    margin: 3rem 0;
    min-height: 400px;
  }

  .performance-summary-toggle-chart {
    display: flex;
    justify-content: center;
  }
}
