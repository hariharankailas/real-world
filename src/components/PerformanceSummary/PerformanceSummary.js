/***********************************************************************************************
 * File : PerformanceSummary
 * Desc : Page to display the performance data
 * Author: Hariharan Kailas
 * Date: 10-Jul-2021
 ***********************************************************************************************/
import React, { useState, useEffect } from 'react';
import MyChart from '../Charts/MyChart';
import MyLineBar from '../Charts/MyLineBar';
import MyAreaChart from '../Charts/MyAreaChart';
import './PerformanceSummary.scss';
const PerformanceSummary = ({ performanceData, clearData }) => {
    const [chartType, setChartType] = useState('');

    useEffect(() => { setChartType('myAreaChart') },
        [performanceData]);
    /***************************************
    * Desc: render charts based on input
    * Date: 08-July-2021
    * ********************************/
    const renderChart = () => {
        switch (chartType) {
            case 'myChart':
                return <MyChart performanceData={ performanceData }/>;
            case 'myLineBar':
                return <MyLineBar performanceData={performanceData}/>;
            case 'myAreaChart':
                return <MyAreaChart performanceData={performanceData}/>;
            default : return <MyAreaChart performanceData={performanceData}/>
        }
    }
    
    return (
        <div className="container performance-summary-container">
            <h1>Performance Summary</h1>
            <div class="row performance-summary-chart">
                {renderChart()}
            </div>
            <div class="row performance-summary-toggle-chart">
                <div class="btn-group" role="group" aria-label="Chart Buttons">
                    <button type="button" id="myAreaChart" class="btn btn-success" onClick={ () => setChartType('myAreaChart')}>Area Chart</button>
                    <button type="button" id="myChart" class="btn btn-primary" onClick={ () => setChartType('myChart')}>Line Chart</button>
                    <button type="button" id="myLineBar" class="btn btn-secondary" onClick={ () => setChartType('myLineBar')}>Bar Chart</button>
                </div>
            </div>
               <div class="row mt-2 d-inline-flex performance-summary-toggle-chart">
                    <button type="button" class="btn btn-danger" onClick={clearData}>Clear Data</button>
            </div>
        </div>
    );
}

export default PerformanceSummary;