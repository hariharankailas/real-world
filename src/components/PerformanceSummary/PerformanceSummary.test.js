import React from 'react';
import { shallow } from 'enzyme';
import PerformanceSummary from './PerformanceSummary';
import MyChart from '../Charts/MyChart';
import MyAreaChart from '../Charts/MyAreaChart';
import MyLineBar from '../Charts/MyLineBar';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container =  shallow(<PerformanceSummary/>);
});

afterEach(() => {
  // cleanup
  container = null;
});

describe('LandingPage', () => {
  it('should render correctly', () => {
    expect(container).toMatchSnapshot();
  });

    it('on click of myAreaChart Button set state chartType to myChart', () => {
    const selectChartType = jest.spyOn(React, "useState");
    const setChartType = jest.fn();
    selectChartType.mockImplementation(chartType => [chartType, setChartType]);

    container.find('#myAreaChart').simulate('click');
    expect(container.find(MyAreaChart)).toHaveLength(1)
  });

  it('on click of myChart Button set state chartType to myChart', () => {
    const selectChartType = jest.spyOn(React, "useState");
    const setChartType = jest.fn();
    selectChartType.mockImplementation(chartType => [chartType, setChartType]);

    container.find('#myChart').simulate('click');
    expect(container.find(MyChart)).toHaveLength(1)
  });

  it('on click of myLineBar Button set state chartType to myChart', () => {
    const selectChartType = jest.spyOn(React, "useState");
    const setChartType = jest.fn();
    selectChartType.mockImplementation(chartType => [chartType, setChartType]);

    container.find('#myLineBar').simulate('click');
    expect(container.find(MyLineBar)).toHaveLength(1)
  });

  
});