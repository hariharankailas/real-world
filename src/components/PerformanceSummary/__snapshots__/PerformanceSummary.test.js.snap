// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render correctly 1`] = `
<div
  className="container performance-summary-container"
>
  <h1>
    Performance Summary
  </h1>
  <div
    class="row performance-summary-chart"
  >
    <MyAreaChart />
  </div>
  <div
    class="row performance-summary-toggle-chart"
  >
    <div
      aria-label="Chart Buttons"
      class="btn-group"
      role="group"
    >
      <button
        class="btn btn-success"
        id="myAreaChart"
        onClick={[Function]}
        type="button"
      >
        Area Chart
      </button>
      <button
        class="btn btn-primary"
        id="myChart"
        onClick={[Function]}
        type="button"
      >
        Line Chart
      </button>
      <button
        class="btn btn-secondary"
        id="myLineBar"
        onClick={[Function]}
        type="button"
      >
        Bar Chart
      </button>
    </div>
  </div>
  <div
    class="row mt-2 d-inline-flex performance-summary-toggle-chart"
  >
    <button
      class="btn btn-danger"
      type="button"
    >
      Clear Data
    </button>
  </div>
</div>
`;
